### PR TITLE
feat(macro/MDNSidebar): rewrite macro + add missing pages

### PR DESCRIPTION
**Pull request by OnkarRuikar**
Add all the pages under https://github.com/mdn/content/tree/main/files/en-us/mdn to the macro.

All the documents under the dir hierarchy have been added. Sequence is as per mentions of the pages in coresponding overview pages.

https://github.com/mdn/yari/assets/87750369/66f28b02-dd33-4eac-bc4b-b2232825f657

### Before change

before

### After changes

after

### CODE REVIEW

File: kumascript/macros/MDNSidebar.ejs [hunk 1]
```diff
@@ -6,244 +6,699 @@ const baseURL = `/${locale}/docs/MDN/`;
...
+        "page_structures_live_examples": "Live examples",
+        "page_structures_page_types": "Page types",
+        "page_structures_quicklinks": "Sidebars and other link macros",
+        "page_structures_specification_tables": "Specification tables",
+        "page_structures_syntax_sections": "Syntax sections",
+        "page_structures_using_macros": "Using macros",
+      "attribution": "Attribution and copyright",
+      "statuses": "Experimental, deprecated and obsolete",
+    "changelog": "Changelog",
+    "kitchensink": "The MDN Content Kitchensink"
```
Comment (teoli2003) [inline kumascript/macros/MDNSidebar.ejs:69]: I don't think we should list this one. It is a technical page used to test MDN.

Comment (OnkarRuikar) [reply kumascript/macros/MDNSidebar.ejs:69]: It could serve as a cheatsheet of all the available constructs. WDYT?

Comment (teoli2003) [reply kumascript/macros/MDNSidebar.ejs:69]: I don't think so: it is not maintained with this goal. And I'm not convinced this additional goal will not conflict with its current tecnhical goal.

**Review by teoli2003 — APPROVED** [review]

**Comment by wbamberg** [comment]
This looks great! Did you consider using the design in https://github.com/mdn/yari/blob/main/kumascript/macros/LearnSidebar.ejs, which derives link text from page titles wherever possible, to avoid these huge string maps for l10n?

**Comment by OnkarRuikar** [comment]
@wbamberg I've updated the macro to use titles. It does render well in local environment. 

But don't know why the `mdnsidebar.test.ts` fails. May be because [content repo is not checked out](https://github.com/mdn/yari/actions/runs/7594299406/workflow?pr=10329) and `.env` is not set. So the code is not able find the page to get the titles. The `learnSidebar.ts` doesn't have test. Should I delete `mdnsidebar.test.ts` file?

**Comment by caugner** [comment]
> Should I delete `mdnsidebar.test.ts` file?

FWIW I think it's okay to remove that test.

**Comment by OnkarRuikar** [comment]
> > Should I delete `mdnsidebar.test.ts` file?
> 
> FWIW I think it's okay to remove that test.

Done in https://github.com/mdn/yari/pull/10329/commits/bbafd2326826cfb7c58223f3d1346770ee6073e3

Also removed from the content kitchensink page: https://github.com/mdn/content/pull/31846

**Comment by wbamberg** [comment]
@caugner, please let me know if you're OK for me to review+merge this PR, or if you want to look.

**Comment by caugner** [comment]
> @caugner, please let me know if you're OK for me to review+merge this PR, or if you want to look.

Let me run this by the content team, to be sure.

**Comment by OnkarRuikar** [comment]
I think we need to wait till https://github.com/mdn/content/issues/31868 is fixed. We'll have to update the macro again after those changes are done.

**Comment by caugner** [comment]
> I think we need to wait till [mdn/content#31868](https://github.com/mdn/content/issues/31868) is fixed.

https://github.com/mdn/content/pull/31878 fixed that meanwhile.

**Comment by OnkarRuikar** [comment]
@caugner I've updated this PR accordingly. We can merge this now.

**Comment by caugner** [comment]
> @caugner I've updated this PR accordingly. We can merge this now.

@OnkarRuikar Thank you. Can you please add Before/After screenshots to the PR description? This will make it easier to have the change reviewed by the content team.

**Comment by OnkarRuikar** [comment]
> @OnkarRuikar Thank you. Can you please add Before/After screenshots to the PR description? This will make it easier to have the change reviewed by the content team.

Done!

**Review by caugner — CHANGES_REQUESTED** [review]
Had a quick chat with the content team, and all looks good, but can we change the order as follows: 🙏 

- Community guidelines
- Writing guidelines
- PAB
- History

**Comment by OnkarRuikar** [comment]
> Had a quick chat with the content team, and all looks good, but can we change the order as follows: 🙏
> 
> * Community guidelines
> * Writing guidelines
> * PAB
> * History

This is how the new order looks:
the new orer

**Review by caugner — APPROVED** [review]

**Pull request merged**